tests/src/core/testqgsproject.cpp

```diff
@@ -884,6 +885,18 @@ void TestQgsProject::testAttachmentsQgz()
 
 }
 
+void TestQgsProject::testAttachmentIdentifier()
+{
+  // Verify attachment identifiers
+  {
+    QgsProject p;
+    QString attachmentFileName = p.createAttachedFile( "test.jpg" );
+    QString attachmentId = p.getAttachmentIdentifier( attachmentFileName );
+    QCOMPARE( p.resolveAttachementIdentifier( attachmentId ), attachmentFileName );
```

Comment:
Would it make sense to add a test that saves the project, reloads and then resolves the identifier to a file path?